tjdashboard/code/site/router.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * @version    CVS: 1.0.0
+ * @package    Com_Tjdashboard
+ * @author     Techjoomla <extensions@techjoomla.com>
+ * @copyright  2017 Techjoomla
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+// No direct access
+defined('_JEXEC') or die;
+
+JLoader::registerPrefix('Tjdashboard', JPATH_SITE . '/components/com_tjdashboard/');
+
+/**
+ * Class TjdashboardRouter
+ *
+ * @since  3.3
+ */
+class TjdashboardRouter extends JComponentRouterBase
+{
+	/**
+	 * Build method for URLs
+	 * This method is meant to transform the query parameters into a more human
+	 * readable form. It is only executed when SEF mode is switched on.
+	 *
+	 * @param   array  &$query  An array of URL arguments
+	 *
+	 * @return  array  The URL arguments to use to assemble the subsequent URL.
+	 *
+	 * @since   3.3
+	 */
+	public function build(&$query)
+	{
+		$segments = array();
+		$view     = null;
+
+		if (isset($query['task']))
+		{
+			$taskParts  = explode('.', $query['task']);
+			$segments[] = implode('/', $taskParts);
+			$view       = $taskParts[0];
+			unset($query['task']);
+		}
+
+		if (isset($query['view']))
+		{
+			$segments[] = $query['view'];
+			$view = $query['view'];
+
+			unset($query['view']);
+		}
+
+		if (isset($query['id']))
+		{
+			if ($view !== null)
+			{
+				$segments[] = $query['id'];
```

Comment:
you should convert the id to the alias when building.. otherwise what's the point of having alias field ??